lightningd/lightningd.c

```diff
@@ -411,51 +600,80 @@ int main(int argc, char *argv[])
 	else if (max_blockheight != UINT32_MAX)
 		max_blockheight -= ld->config.rescan;
 
+	/*~ That's all of the wallet db operations for now. */
 	db_commit_transaction(ld->wallet->db);
 
-	/* Initialize block topology (does its own transaction) */
-	setup_topology(ld->topology, &ld->timers, min_blockheight, max_blockheight);
+	/*~ Initialize block topology.  This does its own io_loop to
+	 * talk to bitcoind, so does its own db transactions. */
+	setup_topology(ld->topology, &ld->timers,
+		       min_blockheight, max_blockheight);
 
-	/* Create RPC socket (if any) */
+	/*~ Create RPC socket (if any): now we can talk to clients. */
```

Comment:
I believe this one is ncessary  to have lightning-cli. Maybe it is standard stuff for people familar with rpc but how is such an API exposed (sure over unix domain socket) but where in the code are api calls provided? I guess this doesnot happen in jsonrpc so a hint in this comment would be nice.

Reply:
Currently the JSON-RPC autoregisters callbacks using the `ccan/autodata` module, however that poses some issues if we'd like to use `clang` based sanitizers since it does weird things with uninitialized data, so maybe we'd want to ween off that anyway at some point.